src/main/config/config.c

```diff
@@ -682,6 +682,20 @@ void validateAndFixGyroConfig(void)
         }
 #endif
 
+#ifdef USE_GYRO_DENOM_CHECK
+        bool bmiDetected = gyro.gyroToUse == 0
+            ? gyro.gyroSensor1.gyroDev.gyroHardware == GYRO_BMI160 || gyro.gyroSensor1.gyroDev.gyroHardware == GYRO_BMI270
+            : gyro.gyroSensor2.gyroDev.gyroHardware == GYRO_BMI160 || gyro.gyroSensor2.gyroDev.gyroHardware == GYRO_BMI270;
+
+        if (bmiDetected) {
+            pidConfigMutable()->pid_process_denom = 1;
+        }
+
+        if (motorConfig()->dev.useDshotTelemetry && rxRuntimeState.rxProvider == RX_PROVIDER_SPI) {
+            pidConfigMutable()->pid_process_denom = bmiDetected ? 2 : 4;
+        }
```

Comment:
What is this part supposed to do? Do we really need the gyro rate to be divided by 4? This doesn't seem right because at this point we don't even know at which looprate the firmware runs at, so "blindly" dividing by 4 can give us undefined behavior.

Reply:
F4[11] with SPI RX and Bidirectional Dshot with a 
- non BMI gyro recommended pidloop of 2Khz
- BMI gyro recommends pidloop of 1.6Khz

MCU | IMU | Sampling Rate | Bidirectional DShot Status | PID Loop Rate | DShot Protocol
-- | -- | -- | -- | -- | --
F7X2, H7XX, G4XX, and similar | MPU60X0, ICM2060X, ICM42688P | 8 kHz | Enabled or Disabled | 8 kHz | DShot 600
  | BMI-270 | 3.2 kHz | Enabled or Disabled | 3.2 kHz | DShot 300
F405 | MPU6000, MPU6050, ICM20601, ICM42688P | 8 kHz | Enabled | 4 kHz | DShot 300
  | MPU6000, MPU6050, ICM20601, ICM42688P | 8 kHz | Disabled (not recommended) | 8 kHz | DShot 600
  | BMI-270 | 3.2 kHz | Enabled or Disabled | 3.2 kHz | DShot 300
F411 UART Rx ** | MPU6000, MPU6050, ICM20601 | 8 kHz | Enabled | 4 kHz | DShot 300
  | MPU6000, MPU6050, ICM20601 | 8 kHz | Disabled (not recommended) | 8 kHz | DShot 600
  | BMI-270 | 3.2 kHz | Enabled | 3.2 kHz | DShot 300
F411 SPI Rx *** | MPU6000, MPU6050, ICM20601 | 8 kHz | Enabled | 2 kHz | DShot 300
  | MPU6000, MPU6050, ICM20601 | 8 kHz | Disabled (not recommended) | 4 kHz | DShot 300
  | BMI-270 | 3.2 kHz | Enabled | 1.6 kHz | DShot 300
  | BMI-270 | 8 kHz | Disabled (not recommended) | 3.2 kHz | DShot 300

Reply:
I guess then we should rather check for the current gyro sample rate and then denominate the PID loop accordingly, instead of checking for specific gyro devices in use.

Reply:
But you are right - we don't need to know which gyro we are using.

if (dshot_telem) denom = 2;
if (sampleRateHz == 8000) denom = (dshotTelem) ? 4 : 2
if (sampleRateHz == 3200) denom = (dshotTelem) ? 2 : 1

target | gyro.sampleRateHz | gyro.sampleLooptime | gyro.targetLooptime | pid-denom | dshot_telem
--- | --- | --- | --- | --- | ---
F405 | 8000 | 125 | 125 | 1 | off
|  |  | 250 | 250 | 2 | on
|  | 3200 | 312,5 | 312,5 | 1 | off/on
F411 UART | 8000 | 125 | 125 | 1 | off
|  |  | 250 | 250 | 2 | on
|  | 3200 | 312.5 | 312.5 | 1 | off/on
F411 SPI | 8000 | 250 | 250 | 2 | off
|  |  | 500 | 500 | 4 | on
|  | 3200 | 312.5 | 312.5 | 1 | off
|  |  | 625 | 625 | 2 | on

Reply:
@SupaflyFPV do you know why we use 1.6Khz in the docs for F411 with SPI RX and Dshot Telemetry enabled?

Reply:
target | gyro.sampleRateHz | gyro.sampleLooptime | pid-denom | dshot_telem
--- | --- | --- | --- | ---
F405 | 8000 | 125 | 1 | off
|  | 4000 | 250 |  2 | on
|  | 3200 (<4000) | 312,5 (>250) | 1 | off/on
F411 UART | 8000 | 125 | 1 | off
|  |  4000 | 250 | 2 | on
|  | 3200 (<4000) | 312.5 (>250) | 1 | off/on
F411 SPI | 4000 | 250 | 2 | off
|  | 3200 (<4000) | 312.5 (>250) | 1 | off
|  | 2000 | 500 | 4 | on
|  | 1600 (<2000) | 625 (>500) | 2 | on